src/quartz/context/param_info.cpp

```diff
@@ -57,6 +57,7 @@ std::vector<ParamType> ParamInfo::get_all_input_param_values() const {
 int ParamInfo::get_new_param_id(const ParamType &param) {
   int id = (int)is_parameter_symbolic_.size();
```

Comment:
I would add an assertion here (and similar for lines 70 and 101 in this file):
```suggestion
  int id = (int)is_parameter_symbolic_.size();
  assert(id == (int)is_parameter_halved_.size());
```